streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractStateManager.java

```diff
@@ -62,11 +63,14 @@ public void reinitializeStateStoresForPartitions(final Logger log,
             checkpointableOffsets.remove(topicPartition);
             storeToBeReinitialized.add(changelogTopicToStore.get(topicPartition.topic()));
         }
-        try {
-            checkpoint.write(checkpointableOffsets);
-        } catch (final IOException fatalException) {
-            log.error("Failed to write offset checkpoint file to {} while re-initializing {}: {}", checkpoint, stateStores, fatalException);
-            throw new StreamsException("Failed to reinitialize global store.", fatalException);
+
+        if (eosEnabled) {
```

Comment:
Sorry for interrupt. I guess you might miss `!` in front of `eosEnabled`. Right ?

Reply:
Great catch! Thanks for reviewing! (this happens when stupid me does not write a unit test...)